### PR TITLE
feature: Release notes for Codacy Self-hosted 3.5.0

### PR DESCRIPTION
**Pull request by machadoit**

### CODE REVIEW

File: docs/release-notes/self-hosted/self-hosted-v3.5.0.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Self-hosted v3.5.0
+
+These release notes are for [Codacy Self-hosted v3.5.0](https://github.com/codacy/chart/releases/tag/3.5.0){: target="_blank"}, released on March 31, 2021.
+
+To upgrade Codacy, follow [these instructions](../../chart/maintenance/upgrade.md).
+
+## Product enhancements
+
+-   Now, Codacy allows [configuring who is allowed to ignore files](https://docs.codacy.com/organizations/roles-and-permissions-for-synced-organizations/#change-analysis-configuration) on all supported Git providers. (CY-3883)
+-   Now, Codacy allows [configuring who is allowed to manage patterns](https://docs.codacy.com/organizations/roles-and-permissions-for-synced-organizations/#change-analysis-configuration) on all supported Git providers.  (CY-3822)
```
Comment (inesgomas) [inline docs/release-notes/self-hosted/self-hosted-v3.5.0.md:10]: I think these should be together like

```suggestion
-   Now, Codacy allows [configuring who is allowed to manage patterns and ignore files](https://docs.codacy.com/organizations/roles-and-permissions-for-synced-organizations/#change-analysis-configuration) on all supported Git providers. (CY-3883)
```

Comment (machadoit) [reply docs/release-notes/self-hosted/self-hosted-v3.5.0.md:10]: Yeah thought about it as well, I like it!

File: docs/release-notes/self-hosted/self-hosted-v3.5.0.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# Self-hosted v3.5.0
+
+These release notes are for [Codacy Self-hosted v3.5.0](https://github.com/codacy/chart/releases/tag/3.5.0){: target="_blank"}, released on March 31, 2021.
+
+To upgrade Codacy, follow [these instructions](../../chart/maintenance/upgrade.md).
+
+## Product enhancements
+
+-   Now, Codacy allows [configuring who is allowed to manage patterns and ignore files](https://docs.codacy.com/organizations/roles-and-permissions-for-synced-organizations/#change-analysis-configuration) on all supported Git providers. (CY-3883)
```
Comment (inesgomas) [inline docs/release-notes/self-hosted/self-hosted-v3.5.0.md:9]: I think the ignore issues to all providers is also being released now. @pedrocodacy @carlospinho04 @franciscodua @josemiguelmelo

Comment (machadoit) [reply docs/release-notes/self-hosted/self-hosted-v3.5.0.md:9]: I think that is not yet included on this release from my understanding, but yeah, if you guys can confirm, would be great! :D

Comment (pedrocodacy) [reply docs/release-notes/self-hosted/self-hosted-v3.5.0.md:9]: Initially we didnt plan to release this on 3.4 but we ended up releasing it. There was a bit of a confusion with that release's notes but they have been patched to introduce this.

tldr: it was released in the previous version

Comment (machadoit) [reply docs/release-notes/self-hosted/self-hosted-v3.5.0.md:9]: Oh ok, thx! I was a bit off about what it was then 😅  either way, we good here then, nice!

File: docs/release-notes/self-hosted/self-hosted-v3.5.0.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+-   PMD 6.28.0
+-   PMD (Legacy) 5.8.1
+-   Prospector 1.3.1
+-   PSScriptAnalyzer 1.18.3
+-   Pylint 1.9.5
+-   **Pylint (Python 3) 2.7.2 (updated from 2.6.0)**
+-   RemarkLint 7.0.1
+-   Revive 1.0.2
+-   **Rubocop 1.12.0 (updated from 1.11.0)**
+-   ScalaStyle 1.5.0
```
Comment (machadoit) [inline docs/release-notes/self-hosted/self-hosted-v3.5.0.md:57]: It seems we ignored this issue on the previous 3.4, and it looks good, so I will also ignore it

File: docs/release-notes/self-hosted/self-hosted-v3.5.0.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
+# Self-hosted v3.5.0
+
+These release notes are for [Codacy Self-hosted v3.5.0](https://github.com/codacy/chart/releases/tag/3.5.0){: target="_blank"}, released on March 31, 2021.
+
+To upgrade Codacy, follow [these instructions](../../chart/maintenance/upgrade.md).
+
+## Product enhancements
+
+-   Now, Codacy allows [configuring who is allowed to manage patterns and ignore files](https://docs.codacy.com/organizations/roles-and-permissions-for-synced-organizations/#change-analysis-configuration) on all supported Git providers. (CY-3883)
+-   Now, Codacy has [Checkov support](https://github.com/bridgecrewio/checkov), extending analysis to infrastructure-as-code like Terraform, Cloudformation and Kubernetes  (CY-3830)
```
Comment (davidscordeiro) [inline docs/release-notes/self-hosted/self-hosted-v3.5.0.md:10]: I want to propose that we rephrase this to highlight the tools/platforms we now support. Something along the lines of:
Codacy is now running Checkov(..), which extends our analysis for infrastructure-as-code: Terraform, Terraform Plan, Cloudformation, Kubernetes, Serverless framework, and ARM Templates are now supported. This new type of analysis contains more than 500 rules, covering compliance and security best practices for AWS, Azure and Google Cloud.

Comment (machadoit) [reply docs/release-notes/self-hosted/self-hosted-v3.5.0.md:10]: Alright, sounds good to me!

**Review by davidscordeiro — APPROVED** [review]
